src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -131,7 +130,11 @@ export class ExamUpdateComponent implements OnInit {
     get isValidExamStudentReviewEnd(): boolean {
         // allow instructors to set examStudentReviewEnd later
         if (!this.exam.examStudentReviewEnd) {
-            return true;
+            if (!this.exam.examStudentReviewStart || !this.exam.examStudentReviewStart.isValid()) {
+                return true;
+            } else {
+                return false;
+            }
```

Comment:
the comment in line 131 does not match the implementation any more or am I wrong? You should change it and describe the "new" logic that you introduced IMO. Maybe this is more readable:
```typescript
    /**
     * checks if a review end date set if there is a valid start date
     */
    get isValidExamStudentReviewEnd(): boolean {
        let isValidExamStudentReviewEnd = false;
        if (this.examStudentReviewStart && this.examStudentReviewStart.isValid() && this.examStudentReviewEnd) {
            isValidExamStudentReviewEnd = true;
        }
        return isValidExamStudentReviewEnd;
    }
```
As you change the logic in this code area, did you consider adapting the client tests? There has to be a case that tests to submit the exam creation form with/without setting start/end date. I would guess that you have to add new test cases, if there were good test cases that would cover the validity of start/end dates, they should have failed with these changes...?

Reply:
Yes, the comment has to be adapted in order to explain the new functionality.
I created some client test cases for date validation that cover the cases of a defined start date but undefined end date (and vice versa) and a start and end date that do not follow the right chronological order in order to test the case Simon already pointed out.
Unfortunately, the proposed code refactoring does not satisfy the test cases, because it does not consider the case that the start date as well as the end date is undefined as valid configuration. Therefore I incorporated the refactoring proposed by @Olimbia23.
Finally, I extended the warning messages in order to make them more general and to cover both cases (missing end date and wrong chronology). I'm not sure whether the message that is displayed when hovering over the warning flag is too long now. Let me know your opinion on this (including, but not limited to @SimonEntholzer @Olimbia23).
Also, feel free to have a look at the added test cases and the newly adapted comment.
Again, thanks for your reviewing and testing effort!